addons/stock/models/stock_move.py

```diff
@@ -2151,15 +2151,19 @@ def _trigger_scheduler(self):
 
         orderpoints_by_company = defaultdict(lambda: self.env['stock.warehouse.orderpoint'])
         orderpoints_context_by_company = defaultdict(dict)
+        orderpoints = self.env['stock.warehouse.orderpoint'].search_fetch([
+            ('product_id', 'in', self.product_id.ids),
+            ('trigger', '=', 'auto'),
+            ('location_id', 'parent_of', self.location_id.ids),
+            ('company_id', 'in', self.company_id.ids),
+        ], ['product_id', 'location_id', 'company_id', 'product_min_qty'])
         for move in self:
-            orderpoint = self.env['stock.warehouse.orderpoint'].search([
-                ('product_id', '=', move.product_id.id),
-                ('trigger', '=', 'auto'),
-                ('location_id', 'parent_of', move.location_id.id),
-                ('company_id', '=', move.company_id.id),
-                '!', ('location_id', 'parent_of', move.location_dest_id.id),
-            ], limit=1)
+            orderpoint = orderpoints.filtered(lambda o: o.product_id == move.product_id and
```

Comment:
maybe we can do a groupby product first to avoid filtering the whole set at each iteration